### DISCUSSION TITLE
[Feature Request] Favicons? Bookmark folders?

### DISCUSSION THREAD

**Discussion by Darthagnon** [go-shiori/shiori#371] [Feature requests]
Would it be possible to download and embed favicons for bookmarks (maybe even set our own favicons)? The text-based boxes sorta blur together and I can't find anything visually. I also really miss the folder layout of my bookmarks, too (related issue? https://github.com/go-shiori/shiori/issues/125)...

**Comment by fmartingr** [comment]
Hey @Darthagnon, I've moved the issue to the feature request forums so people can discuss and vote on it.

**Reply by Darthagnon** [reply]
Thank you!

**Comment by GeniusBroccoli** [comment]
I was looking for a bookmark manager and accidentally found Shiori. The first impression is very positive! But really, I'd like to see folders and maybe a table-like view - it's pretty unusual, but I use it when one source has multiple links to different sites. More advanced users might argue by saying that there are tags, but for me, and I think many users, the visual folder view is more comfortable and friendly. And + for favicons.

**Reply by fmartingr** [reply]
I've found out over the years that tags and folders are not so different, and the only thing that matters is how data is presented to the user. If I let nested tags to be a thing, you'll have the same functionality as folders but with a different name, essentially. 

I'm still figuring out things like this. I'm curious, would a better way of using tags be a good compromise to you? I agree that the current way we handle tags is not very user friendly, so maybe doing that more usable would work for people that want folders?

**Reply by GeniusBroccoli** [reply]
Hmm, I think for users that want folders, it's more important to instantly cover all the information visually. Maybe, if you don't make folders, create some kind of tag cloud page where all the tags will be immediately visible and by clicking on the tag the corresponding entries will be sorted nearby or in a separate window. For example, a cloud in the form of cards with the name of the tag and the number of entries.

**Reply by woofie56** [reply]
@fmartingr : 

> If I let nested tags to be a thing

This seemed like a good idea. Did you put it on the roadmap?

**Comment by Katarn** [comment]
Indeed, I often thought that it would be nice to have both. When the bookmark database is large, this would be a good addition for tags. Now, however, most of all there is not enough filtering by domain name. I probably have 300-400 bookmarks from one site, I need to download them to the archive, but they are mixed with other bookmarks. They are quite difficult to filter out.

**Comment by marco-trovato** [comment]
I have 25 years of bookmarks hierarchically sorted in folders.

Please see image for visual reference:
![hierarchy](https://github.com/go-shiori/shiori/assets/18162107/710826e7-b8b4-45e4-8f8e-7c477f6d65b5)

**Shiori is the best bookmark manager i have ever seen**, mainly because it's a portable single executable compilable everywhere, and has both CLI and web interface!
But when importing my bookmarks all i get is 350 pages of this unusable mess with TAGs:
![result](https://github.com/go-shiori/shiori/assets/18162107/0ec88e7a-9d7b-4324-9692-2adaf4dbda9c)

Please add folders/hierarchy for engineers who are used to sort and organize their bookmarks without putting everything in a single melting pot folder.

**Comment by LLKoder** [comment]
IMHO, it would be nice if Shiori have tabs and folders.

Folders provides hierarchy structure of bookmarks.
Tags provides cross links of bookmarks.
These parameters complement each other.

IMHO mix folders and tags are not good idea. I mean don't need to try to do folders from tags. It's confuse users. It's are good variant to save both.

**Comment by fmartingr** [comment]
Hey guys, bookmark folders have been promoted to the Roadmap.

Closing in favor of #1077
